samples/extensions/vertex_dynamic_state/vertex_dynamic_state.cpp

```diff
@@ -0,0 +1,634 @@
+
+	VkPipelineLayoutCreateInfo pipeline_layout_create_info =
+	    vkb::initializers::pipeline_layout_create_info(
+	        &descriptor_set_layout,
+	        1);
+
+	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &pipeline_layout_create_info, nullptr, &pipeline_layout));
+}
+
+/**
+ * 	@fn void vertex_dynamic_state::create_descriptor_sets()
+ * 	@brief Creating both descriptor set:
+ * 		   1. Uniform buffer
+ * 		   2. Image sampler
+*/
+void vertex_dynamic_state::create_descriptor_sets()
+{
+	VkDescriptorSetAllocateInfo alloc_info =
+	    vkb::initializers::descriptor_set_allocate_info(
+	        descriptor_pool,
+	        &descriptor_set_layout,
+	        1);
+
+	VK_CHECK(vkAllocateDescriptorSets(get_device().get_handle(), &alloc_info, &descriptor_set));
+
+	VkDescriptorBufferInfo            matrix_buffer_descriptor     = create_descriptor(*ubo);
+	VkDescriptorImageInfo             environment_image_descriptor = create_descriptor(textures.envmap);
+	std::vector<VkWriteDescriptorSet> write_descriptor_sets        = {
+        vkb::initializers::write_descriptor_set(descriptor_set, VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 0, &matrix_buffer_descriptor),
+        vkb::initializers::write_descriptor_set(descriptor_set, VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, 1, &environment_image_descriptor),
+    };
+	vkUpdateDescriptorSets(get_device().get_handle(), static_cast<uint32_t>(write_descriptor_sets.size()), write_descriptor_sets.data(), 0, nullptr);
+}
+
+void vertex_dynamic_state::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+	/* Enable extension features required by this sample
+	   These are passed to device creation via a pNext structure chain */
+	auto &requested_vertex_input_features                   = gpu.request_extension_features<VkPhysicalDeviceVertexInputDynamicStateFeaturesEXT>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VERTEX_INPUT_DYNAMIC_STATE_FEATURES_EXT);
+	requested_vertex_input_features.vertexInputDynamicState = VK_TRUE;
+
+	if (gpu.get_features().samplerAnisotropy)
+	{
+		gpu.get_mutable_requested_features().samplerAnisotropy = true;
+	}
+}
+
+/**
+ * 	@fn void vertex_dynamic_state::draw_created_model(VkCommandBuffer commandBuffer)
+ * 	@brief Drawing created model using indexing buffer
+ */
+void vertex_dynamic_state::draw_created_model(VkCommandBuffer commandBuffer)
+{
+	VkDeviceSize offsets[1] = {0};
+	vkCmdBindVertexBuffers(commandBuffer, 0, 1, cube.vertices->get(), offsets);
+	vkCmdBindIndexBuffer(commandBuffer, cube.indices->get_handle(), 0, VK_INDEX_TYPE_UINT32);
+	vkCmdDrawIndexed(commandBuffer, cube.index_count, 1, 0, 0, 0);
+}
+
+/**
+ *  @fn void vertex_dynamic_state::model_data_creation()
+ *  @brief Generate vertex input data for simple cube (position and normal vectors)
+ */
+void vertex_dynamic_state::model_data_creation()
+{
+	constexpr uint32_t vertex_count = 8;
+	SampleVertex *     vertices     = new SampleVertex[vertex_count];
```

Comment:
Why aren't you using a simple C-Array here (`SampleVertex vertices[vertex_count];`)? Or maybe a std::vector or a std::array?

Reply:
It's because of my habits (I works before mostly in C). I will change it to something more c++